src/gfx.cpp

```diff
@@ -1059,18 +1059,23 @@ void DrawSprite(SpriteID img, PaletteID pal, int x, int y, const SubSprite *sub,
 /**
  * The code for setting up the blitter mode and sprite information before finally drawing the sprite.
  * @param sprite The sprite to draw.
- * @param x      The X location to draw.
- * @param y      The Y location to draw.
- * @param mode   The settings for the blitter to pass.
- * @param sub    Whether to only draw a sub set of the sprite.
- * @param zoom   The zoom level at which to draw the sprites.
+ * @param x The X location to draw.
+ * @param y The Y location to draw.
+ * @param mode The settings for the blitter to pass.
+ * @param sub Whether to only draw a sub set of the sprite.
+ * @param zoom The zoom level at which to draw the sprites.
+ * @param dst Optional parameter for a different blitting destination.
  * @tparam ZOOM_BASE The factor required to get the sub sprite information into the right size.
  * @tparam SCALED_XY Whether the X and Y are scaled or unscaled.
  */
 template <int ZOOM_BASE, bool SCALED_XY>
-static void GfxBlitter(const Sprite * const sprite, int x, int y, BlitterMode mode, const SubSprite * const sub, SpriteID sprite_id, ZoomLevel zoom)
+static void GfxBlitter(const Sprite * const sprite, int x, int y, BlitterMode mode, const SubSprite * const sub, SpriteID sprite_id, ZoomLevel zoom, const DrawPixelInfo *dst = nullptr)
 {
 	const DrawPixelInfo *dpi = _cur_dpi;
+	if (dst != nullptr) {
+		dpi = dst;
+	}
```

Comment:
`dpi = (dst != nullptr) ? dst : _cur_dpi`
maybe?
Makes it slightly easier to read, all those 3 letter variables are hard to keep apart :)